### ISSUE TITLE
Add cpp-ethereum in Peter's hive tests

### ISSUE THREAD

**Issue by bobsummerwill** [ethereum/cpp-ethereum#3203]
_From @bobsummerwill on July 18, 2016 4:40_

See https://github.com/karalabe/hive
# hive - Ethereum end-to-end test harness

Ethereum grew large to the point where testing implementations is a huge burden. Unit tests are fine for checking various implementation quirks, but validating that a client conforms to some baseline quality or validating that clients can play nicely together in a multi client environment is all but simple.

This project is meant to serve as an easily expandable test harness where anyone can add tests in any programming language felt comfortable with and it should simultaneously be able to run those tests against all potential clients. As such the harness is meant to do black box testing where no client specific internal details/state can be tested and/or inspected, rather emphasis being put on adherence to official specs or behaviours under different circumstances.

Most importantly, it is essential to be able to run this test suite as part of the CI workflow! To this effect the entire suite is based on docker containers.

@karalabe said ..

"If you guys/gals an interested in getting the C++ client in there, I'd gladly help 
Basically this is all that's needed for a client to be tested https://github.com/karalabe/hive/tree/master/clients/go-ethereum:master"

_Copied from original issue: ethereum/webthree-umbrella#651_

**Comment by winsvega** [comment]
PR to work with 
https://github.com/karalabe/hive/pull/17

**Comment by chfast** [comment]
@holiman is this done?

**Comment by holiman** [comment]
It's done!

**Comment by bobsummerwill** [comment]
Awesome, @holiman!

Is there a publicly-visible dashboard or CI setup for Hive?

**Comment by holiman** [comment]
Not yet, but here's a preview: http://martin.swende.se/foobar/report-2017-02-23_15-21-19/ 
I'll be moving it to automated execution within azure in ~two weeks

**Comment by bobsummerwill** [comment]
Great!   Thanks for the link, @holiman :-)   Best wishes.
